### DISCUSSION TITLE
More details on IPoverWB (IPv4 over Wifibroadcast tunnel)?

### DISCUSSION THREAD

**Discussion by Mysnomer0** [svpcom/wfb-ng#103] [Questions]
Hi @svpcom ,

I'm confused on how to setup IPoverWB mentioned at the end of the Setup HOWTO wiki page. After doing some research, I'm still confused on how to setup and install a tun/tap kernel driver for Ubuntu. I understand what a tun/tap kernel driver does in theory, but how to install and use it in this application is still unclear. Do you have some references you could point me to? Or, perhaps, a list of commands to run? If I'm able to set it up, I'll try to contribute to the wiki a more detailed guide on setting it up.

Thanks,

Milan

**Comment by SummerSeaSun** [comment]
Hi, 
I'm interested to an example would be great. Or at least the list of steps to make this work.
1 - set  ip address on the machines:
`sudo ifconfig eth0 "sender" netmask 255.255.255.0 up`
`sudo ifconfig eth0 "receiver" netmask 255.255.255.0 up`
2 - where you specify sender and receiver addresses in config?
Thanks, 
BR

**Comment by Mysnomer0** [comment]
Hi BR,

I'm sorry, I'm confused by your comment. Are you also wanting the answer to my question, or are you trying to answer it yourself?

Thanks,

Milan

**Comment by svpcom** [comment]
For IPoverWB (IPv4 over Wifibroadcast tunnel) you need only tun/tap kernel driver (tun.ko). Tunnel will be configured and work out of box. Drone side will have 10.5.0.2/24 address and GS - 10.5.0.1/24.
So you need:
1. Install .deb package on the both machines
2. Edit `/etc/default/wifibroadcast` and set `WFB_NICS="wlanXXX"`
3. Run `sudo systemctl restart wifibroadcast@gs` on the first and `sudo systemctl restart wifibroadcast@drone` on the second
4. Use `ifconfig` or `ip link ls` to check tunnel interface

**Comment by svpcom** [comment]
Builtin config file with all default values: /usr/lib/python2.7/dist-packages/telemetry/conf/master.cfg
Create /etc/wifibroadcast.cfg if you need to override some of them

**Comment by appetito** [comment]
Hello! I have tunel iface up but cant ssh from gs to drone:

gs-wfb: flags=4305<UP,POINTOPOINT,RUNNING,NOARP,MULTICAST>  mtu 1446
        inet 10.5.0.1  netmask 255.255.255.0  destination 10.5.0.1
        inet6 fe80::6c97:523c:6ac3:fef8  prefixlen 64  scopeid 0x20<link>
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 500  (UNSPEC)
        RX packets 1  bytes 48 (48.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 27  bytes 1704 (1.6 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

drone-wfb: flags=4305<UP,POINTOPOINT,RUNNING,NOARP,MULTICAST>  mtu 1446
        inet 10.5.0.2  netmask 255.255.255.0  destination 10.5.0.2
        inet6 fe80::c585:5208:8af4:6068  prefixlen 64  scopeid 0x20<link>
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 500  (UNSPEC)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 9  bytes 432 (432.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

┌──[tunnel]────────────────────────────────────────────────────────────────────────────────────────────────────┐
│TX ANT 00**                                                                                                   │
│RX PKT:   recv 2568 d_ok 2568 fec_r 280 lost 835 d_err 0 bad 0                                                │
│RX PKT/s: recv 2 d_ok 2 fec_r 1 lost 1 d_err 0 bad 0                                                          │
│                                                                                                              │
│0001: 2 pkt/s, rssi -3 < -3 < -3

**Comment by Mysnomer0** [comment]
@svpcom Thanks a ton for your input, I'm sorry I didn't get back to you. I'll try out your suggestions as soon as I can.
